src/DateTimeParser.jl

```diff
@@ -242,17 +239,19 @@ function _parsedate(datetimestring::AbstractString; fuzzy::Bool=false,
                 end
             elseif i+1 <= len && tokens[i] == ":"
                 # HH:MM[:SS[.ss]]
-                res["hour"] = parse(Int, token)
-                res["minute"] = parse(Int, tokens[i+1])
+                res["hour"] = Hour(token)
+                res["minute"] = Minute(tokens[i+1])
                 i += 2
                 if i+1 <= len && tokens[i] == "." && isdigit(tokens[i+1])
-                    res["second"] = round(Int, 60 * parse(Float64, string(".", tokens[i+1])))
+                    temp = 60 * parse(Float64, string(".", tokens[i+1]))
+                    res["second"] = Second(round(Int, temp))
                     i += 2
                 elseif i < len && tokens[i] == ":"
-                    res["second"] = parse(Int, tokens[i+1])
+                    res["second"] = Second(tokens[i+1])
                     i += 2
                     if i+1 <= len && tokens[i] == "." && isdigit(tokens[i+1])
-                        res["millisecond"] = round(Int, 1000 * parse(Float64, string(".", tokens[i+1])))
+                        temp = 1000 * parse(Float64, string(".", tokens[i+1]))
+                        res["millisecond"] = Millisecond(round(Int, temp))
                         i += 2
                     end
                 end
```

Comment:
Ending with a decimal is a special case. I'm not sure we want to support having a floats in the middle.